src/commands.ts

```diff
@@ -27,6 +29,12 @@ const commands = {
         vscode.tasks.executeTask(task);
     },
 
+    makeDebugConfig(ctx: SwiftContext): Command {
```

Comment:
Perhaps we don't need this factory if we can just use argument binding instead.

Can you try defining the command as:

```typescript
async makeDebugConfig(ctx: SwiftContext) {
    return await debug.makeDebugConfigurations(ctx);
},
```

and then registering it as:

```typescript
vscode.commands.registerCommand('swift.makeDebugConfig',
  this.makeDebugConfig.bind(this, ctx)
)
```

This works for me, but I'm not sure it's 100% what you want.

Reply:
Didn't know about the argument binding, will change.